pkg/updater/download_test.go

```diff
@@ -6,322 +6,132 @@
 
-	header.Name = filename
-
-	err = tw.WriteHeader(header)
+func (s *testFixturesServer) PackageFS(f fixture) fs.FS {
+	fs, err := fs.Sub(fixturesFS, f.contentPath)
 	if err != nil {
-		return err
+		panic(err)
 	}
+	return fs
+}
 
-	_, err = io.Copy(tw, file)
+func (s *testFixturesServer) Image(f fixture) oci.Image {
+	tmpDir := s.t.TempDir()
+	image, err := s.Downloader().Download(context.Background(), tmpDir, s.Package(f))
 	if err != nil {
-		return err
+		panic(err)
 	}
-
-	return nil
+	return image
 }
 
-func (suite *DownloadTestSuite) TestDownload() {
-	t := suite.T()
-	downloader := newDownloader(suite.server.Client())
-	downloadPath := path.Join(suite.dir, testDownloadDir)
-
-	pkg := Package{
-		Name:    "datadog-agent",
-		Version: "7.51.0",
-		URL:     fmt.Sprintf("%s/%s", suite.server.URL, testAgentArchiveFileName),
-		SHA256:  agentArchiveHash(t, suite.dir),
-		Size:    agentArchiveSize(t, suite.dir),
+func (s *testFixturesServer) Catalog() catalog {
+	return catalog{
+		Packages: []Package{
+			s.Package(fixtureSimpleV1),
+			s.Package(fixtureSimpleV2),
+		},
 	}
-	err := downloader.Download(context.Background(), pkg, downloadPath)
-
-	// Read contents of downloadPath
-	assert.NoError(t, err)
-	assert.FileExists(t, path.Join(downloadPath, testAgentFileName))
-	assert.FileExists(t, path.Join(downloadPath, testNestedAgentFileName))
-	assert.FileExists(t, path.Join(downloadPath, testLargeFileName))
+}
 
-	// ensures the full archive or full individual files are not loaded in memory
-	var m runtime.MemStats
-	runtime.ReadMemStats(&m)
-	assert.Less(t, m.TotalAlloc, uint64(testLargeFileSize))
+func (s *testFixturesServer) Close() {
+	s.s.Close()
 }
 
-func (suite *DownloadTestSuite) TestDownloadCheckHash() {
-	t := suite.T()
-	downloader := newDownloader(suite.server.Client())
-	downloadPath := path.Join(suite.dir, testDownloadDir)
-	err := os.MkdirAll(downloadPath, 0755)
-	assert.NoError(t, err)
+func TestDownload(t *testing.T) {
+	s := newTestFixturesServer(t)
+	defer s.Close()
+	d := s.Downloader()
 
-	fakeHash := sha256.Sum256([]byte(`test`))
-	pkg := Package{
-		Name:    "datadog-agent",
-		Version: "7.51.0",
-		URL:     fmt.Sprintf("%s/%s", suite.server.URL, testAgentArchiveFileName),
-		SHA256:  hex.EncodeToString(fakeHash[:]),
-		Size:    agentArchiveSize(t, suite.dir),
-	}
-	err = downloader.Download(context.Background(), pkg, downloadPath)
-	assert.Error(t, err)
+	image, err := d.Download(context.Background(), t.TempDir(), s.Package(fixtureSimpleV1))
+	assert.NoError(t, err)
+	tmpDir := t.TempDir()
+	err = extractPackageLayers(image, tmpDir)
+	assert.NoError(t, err)
+	assertEqualFS(t, s.PackageFS(fixtureSimpleV1), os.DirFS(tmpDir))
 }
 
-func (suite *DownloadTestSuite) TestDownloadCheckSize() {
-	t := suite.T()
-	downloader := newDownloader(suite.server.Client())
-	downloadPath := path.Join(suite.dir, testDownloadDir)
-	err := os.MkdirAll(downloadPath, 0755)
-	assert.NoError(t, err)
+func TestDownloadInvalidHash(t *testing.T) {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏Can you add an invalid platform test?

Reply:
done